vnext/Microsoft.ReactNative/Views/Image/Effects.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#define EFFECTS_NAMESPACE winrt::Microsoft::ReactNative
+#include <BorderEffect.g.h>
+#include <GaussianBlurEffect.g.h>
+
+#pragma warning(push)
+#pragma warning(disable : 28285 28196 6387 6319 26812)
+
+#include "Microsoft.UI.Composition.Effects_Impl.h"
+
+namespace EFFECTS_NAMESPACE::implementation {
+//-----------------------------------------------------------------------------------------------------------------
+class BorderEffect
+    : public BorderEffectT<BorderEffect, winrt::Microsoft::UI::Composition::Effects::implementation::EffectBase> {
+ public:
+  DECLARE_D2D_GUID(CLSID_D2D1Border);
+  DECLARE_SINGLE_SOURCE(Source);
+  DECLARE_POD_PROPERTY(
+      ExtendX,
+      winrt::Microsoft::ReactNative::CanvasEdgeBehavior,
+      winrt::Microsoft::ReactNative::CanvasEdgeBehavior::Clamp,
+      true);
+  DECLARE_POD_PROPERTY(
+      ExtendY,
+      winrt::Microsoft::ReactNative::CanvasEdgeBehavior,
+      winrt::Microsoft::ReactNative::CanvasEdgeBehavior::Clamp,
+      true);
+  DECLARE_NAMED_PROPERTY_MAPPING(
+      {L"ExtendX", D2D1_BORDER_PROP_EDGE_MODE_X, PropertyMapping::GRAPHICS_EFFECT_PROPERTY_MAPPING_DIRECT},
+      {L"ExtendY", D2D1_BORDER_PROP_EDGE_MODE_Y, PropertyMapping::GRAPHICS_EFFECT_PROPERTY_MAPPING_DIRECT});
+
+ public:
+  IFACEMETHODIMP GetPropertyCount(_Out_ UINT *count) override {
+    *count = 2;
+    return S_OK;
+  }
+
+  IFACEMETHODIMP GetProperty(UINT index, _Outptr_ abi::IPropertyValue **value) override {
+    return UsePropertyFactory(value, [=]() {
```

Comment:
Nit: implict capture by value (`[=]`) can be dangerous, or lead to unexpected copies. We should generally do explicit capture, or implicit capture by reference if everything is synchronous.

Reply:
Same below

Reply:
See http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f52-prefer-capturing-by-reference-in-lambdas-that-will-be-used-locally-including-passed-to-algorithms for the ref part